doc/en/user/source/filter/ecql_reference.rst

```diff
@@ -193,8 +193,9 @@ Attribute
 
 An attribute name denotes the value of a feature attribute.
 
-* Simple attribute names are sequences of letters and numbers,
-* Attribute names quoted with double-quotes may be any sequence of characters.
+* Simple attribute names are sequences of letters and numbers, e.g. `States123`
+* Attribute names quoted with double-quotes may be any sequence of characters, e.g. `"States!@#"`
+* `Note <https://gis.stackexchange.com/a/475826/68995>`_: `id` is a reserved word in the GML spec (for the FeatureID) and thus an attribute (or database column) named `id` must be quoted, e.g. `"id"`
```

Comment:
There is some confusion here... "id" and "name" are indeed default attributes of the abstract GML feature (the base "class" for all the features), but the reason why you need to quote "id" in ECQL, is that it's a keyword in that language:

https://github.com/geotools/geotools/blob/main/modules/library/cql/src/main/jjtree/ECQLGrammar.jjt#L180

Reply:
Ah, thank you.  I will update.

Reply:
Checking that reserved word is only used for deprecated syntax:

> <id predicate> ::= "ID" "IN" <id list> 

It does not seem to offer a lot of value, but I understand the difficulty in removing.